Add documentation on Stalker

jonwilson030981 opened the pull request.

jonwilson030981 commented on _i18n/en/_docs/stalker.md:
```diff
@@ -0,0 +1,1859 @@
+
+
+## Options
+Now lets look at the options when we follow a thread with stalker. Stalker generates events when a followed thread is being executed, these are placed onto a queue and flushed either periodically or when the queue is full. The size and time period can be configured by the options. Events can be generated for on a per-instruction basis either for calls, returns or all instructions. Or they can be  ...
```
How does a compile event differ from the event and block events? Presumably, it is when the block is instrumented, rather then when it is run? What circumstances are there where you may instrument a block, but then not execute it (so that one event doesn't occur immediately after the other)?

Cheers.

oleavr replied:
> How does a compile event differ from the event and block events?

The `compile` event is emitted right after a basic block has been compiled, i.e. just before it's executed for the first time. So it's only done once (per thread). This makes it very useful for measuring coverage. (However, if the code mutates it will happen again for the same basic block.)

The `block` event is emitted each time a given basic block is executed, *before* any of its instructions have been executed.

> Presumably, it is when the block is instrumented, rather then when it is run?

Correct.

> What circumstances are there where you may instrument a block, but then not execute it (so that one event doesn't occur immediately after the other)

None that I can think of.

Cheers!

jonwilson030981 commented:
I have made fixed the line lengths. Only thing that is longer now is some of the URLs, but they won't be rendered anyway? Is there anything more to be done for these?

I have added a newline after all of the headings and removed any instances of multiple concurrent empty lines.

Please let me know if my commit will show up automatically in the pull request, or if there's something else you need me to do.

oleavr commented:
> I have made fixed the line lengths. Only thing that is longer now is some of the URLs, but they won't be rendered anyway? Is there anything more to be done for these?

Awesome! No those are fine.

Will go ahead and review the rest now! 👍

jonwilson030981 commented on _i18n/en/_docs/stalker.md:
```diff
@@ -0,0 +1,2620 @@
+Therefore when we want to read a register, this can be achieved with the single
+function `gum_exec_ctx_load_real_register_into`. This determines which kind of
+prologue is in use and calls the relevant routine accordingly. Note that these
+routines don't actually read the registers, they emit code which reads them.
```
Descriptive names are hard though! Either that or you end up with a very long one!

jonwilson030981 commented on _i18n/en/_docs/stalker.md:
```diff
@@ -0,0 +1,2620 @@
+tracks how many instructions have passed since. It seems that this is continued
+for up to four instructions, presumably this was determined by empirical
+testing, although it does make some sense as this is about how many instructions
+would be needed to load, test, modify and store the value. This is then used to
```
Yeah, I guess it might be good to kick out a warning or something if a callout is encountered in such a section.

jonwilson030981 commented:
Ok. I think that's got them. There were a few higher up the pull request which said they were outdated, maybe you were raising them as I was addressing the new lines, or maybe it just got confused when I did some edits in the webpage and some in the editor. Anyways, I think I went back through and fixed them all.

I've also fixed up the latest ones. Please do let me know if I have missed anything though. Or feel free to jump in and just clobber anything you think needs correcting.

oleavr commented:
Thanks! ❤️ Merged 🎉
